lib/kubernetes-deploy/runner.rb

```diff
@@ -290,30 +290,23 @@ def confirm_context_exists
     end
 
     def confirm_namespace_exists
-      _, _, st = run_kubectl("get", "namespace", @namespace, namespaced: false)
+      _, _, st = run_kubectl("get", "namespace", @namespace, with_namespace: false)
```

Comment:
Sometimes we want to run `kubectl` with a namespace flag and sometimes without it:

```
kubectl get pods --namespace blah # get pods of a namespace
kubectl get namespaces # don't need the namespace flag to get all namespaces
```

The same thing applies to `context`.

`with_context` and `with_namespace` take boolean values. I made them `with_` to differentiate that this is a opt-in/opt-out flag, and not the flag that tells what namespace you want to use. Otherwise developers may assume that the flag accepts the namespace name: `run_kubectl("get", "pods", namespace: "mynamespace")`

I'm open to rename them to something like `append_namespace` to address the concern about `with*` methods working with a block.
Does it make sense to you?